pkg/api/common/helper.go

```diff
@@ -351,3 +353,69 @@ func WrapAsParameter(s string) string {
 func WrapAsVerbatim(s string) string {
 	return fmt.Sprintf("',%s,'", s)
 }
+
+// GetDockerConfig transforms the default docker config with overrides. Overrides may be nil.
+func GetDockerConfig(opts map[string]string, overrides []func(*DockerConfig) error) (string, error) {
+	config := GetDefaultDockerConfig()
+
+	for i := range overrides {
+		if err := overrides[i](&config); err != nil {
+			return "", err
+		}
+	}
+
+	dataDir, ok := opts[ContainerDataDirKey]
+	if ok {
+		config.DataRoot = dataDir
+	}
+
+	b, err := json.MarshalIndent(config, "", "    ")
+	return string(b), err
+}
+
+// GetContainerdConfig transforms the default containerd config with overrides. Overrides may be nil.
+func GetContainerdConfig(opts map[string]string, overrides []func(*ContainerdConfig) error) (string, error) {
+	config := GetDefaultContainerdConfig()
+
+	for i := range overrides {
+		if err := overrides[i](&config); err != nil {
+			return "", err
+		}
+	}
+
+	dataDir, ok := opts[ContainerDataDirKey]
+	if ok {
+		config.Root = dataDir
+	}
+
+	buf := new(bytes.Buffer)
+	err := toml.NewEncoder(buf).Encode(config)
+	return buf.String(), err
+}
+
+// ContainerdKubenetOverride transforms a containerd config to set details required when using kubenet.
+func ContainerdKubenetOverride(config *ContainerdConfig) error {
+	config.Plugins.IoContainerdGrpcV1Cri.CNI.ConfTemplate = "/etc/containerd/kubenet_template.conf"
+	return nil
+}
+
+// ContainerdSandboxImageOverrider produces a function to transform containerd config by setting the SandboxImage.
+func ContainerdSandboxImageOverrider(image string) func(*ContainerdConfig) error {
```

Comment:
This is an `Overrider` rather than just an `Override` func so we can use "curried" funcs with an `image` arg baked in elsewhere?

Reply:
That was my thinking, the semantics are already fairly clear so I'm happy to keep the naming consistent (`Override`) if you think it makes sense